lib/_http_server.js

```diff
@@ -237,48 +237,50 @@ ServerResponse.prototype._implicitHeader = function _implicitHeader() {
   this.writeHead(this.statusCode);
 };
 
-ServerResponse.prototype.writeHead = writeHead;
-function writeHead(statusCode, reason, obj) {
+ServerResponse.prototype.writeHead = function(
+  statusCode,
+  statusMessage,
+  headers
+) {
   const originalStatusCode = statusCode;
 
   statusCode |= 0;
   if (statusCode < 100 || statusCode > 999) {
     throw new ERR_HTTP_INVALID_STATUS_CODE(originalStatusCode);
   }
 
-
-  if (typeof reason === 'string') {
-    // writeHead(statusCode, reasonPhrase[, headers])
-    this.statusMessage = reason;
-  } else {
+  if (headers === undefined && typeof statusMessage === 'object') {
     // writeHead(statusCode[, headers])
-    if (!this.statusMessage)
-      this.statusMessage = STATUS_CODES[statusCode] || 'unknown';
-    obj = reason;
+    headers = statusMessage;
+  } else if (typeof statusMessage === 'string') {
+    // writeHead(statusCode, reasonPhrase[, headers])
+    this.statusMessage = statusMessage;
   }
+  if (!this.statusMessage)
+    this.statusMessage = STATUS_CODES[statusCode] || 'unknown';
```

Comment:
This will also cause `res.writeHead(200, '')` to use the default statusMessage, whereas before it would keep the blank string. 

I don't know if this is desired or not.

Reply:
Allow to pass message which is empty string must be a wrong design.

Reply:
How do you figure? A blank status message (aka, `reason-phrase`) is valid HTTP.  See https://tools.ietf.org/html/rfc7230#section-3.1.2

Reply:
@spazmodius is correct. An empty reason phrase is perfectly legal per spec. Using `if (this.statusMessage == null)` would be better here

Reply:
@jasnell thanks for the suggestion.
I forget the magic js `!'' === true`